client/src/search/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,215 @@
+import MediaQuery from "react-responsive";
+
+import { DropdownMenu } from "src/components/index.js";
+
+import { log_standard_event } from "../../core/analytics.js";
+import { breakpoints } from "../../core/breakpoint_defs.js";
+
+import { IconFilter } from "../../icons/icons.js";
+import { create_text_maker } from "../../models/text.js";
+
+import { get_static_url } from "../../request_utils.js";
+
+import { InfoBaseHighlighter } from "../search_utils.js";
+
+import { TypeaheadMenu } from "./TypeaheadMenu.js";
+
+import text from "./Typeahead.yaml";
+
+const text_maker = create_text_maker(text);
+
+import "./Typeahead.scss";
+
+export class Typeahead extends React.Component {
+  state = {
+    search_text: "",
+    can_show_menu: false,
```

Comment:
`can_show_menu`? or `show_menu`?

Reply:
can_show_menu since the length of search text also affects whether menu shows or not